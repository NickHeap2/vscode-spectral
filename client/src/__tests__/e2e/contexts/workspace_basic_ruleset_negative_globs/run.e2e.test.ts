/* eslint-disable require-jsdoc */

import { expect } from 'chai';
import * as chaiJestSnapshot from 'chai-jest-snapshot';
import * as vscode from 'vscode';
import * as path from 'path';

import { openFile, activate, setRulesetFile, setValidateFiles } from '../../helper';
import { workspace } from 'vscode';

suiteSetup(async () => {
  chaiJestSnapshot.resetSnapshotRegistry();
  setRulesetFile('');
  setValidateFiles([
    '**/*.yaml',
    '!**/simple-ignore.yaml',
    '!**/package.json',
    '!**/template.cfn.yaml',
  ]);
  await activate();
});

setup(function() {
  // eslint-disable-next-line no-invalid-this
  chaiJestSnapshot.configureUsingMochaContext(this);
});

suite('Workspace, basic ruleset negative globs', () => {
  suite('Invalid files trigger generation of diagnostics but only if not ignored', () => {
    ['simple.yaml', 'simple.json', 'package.json', 'template.cfn.yaml', 'simple-ignore.yaml'].forEach((fixture) => {
      test(`${fixture}`, async () => {
        const diags = await lint(['invalid', fixture]);

        expect(diags).to.matchSnapshot();
      });
    });
  });

  const lint = async (pathSegments: string[]) => {
    const docPath = path.resolve(workspace.rootPath as string, ...pathSegments);

    const docUri = vscode.Uri.file(docPath);
    await openFile(docUri);

    return vscode.languages.getDiagnostics(docUri);
  };
});
