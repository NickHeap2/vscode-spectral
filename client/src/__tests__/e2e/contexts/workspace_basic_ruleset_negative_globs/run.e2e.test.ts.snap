// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace, basic ruleset negative globs Invalid files trigger generation of diagnostics but only if not ignored package.json 1`] = `Array []`;

exports[`Workspace, basic ruleset negative globs Invalid files trigger generation of diagnostics but only if not ignored simple.json 1`] = `Array []`;

exports[`Workspace, basic ruleset negative globs Invalid files trigger generation of diagnostics but only if not ignored simple.yaml 1`] = `
Array [
  Object {
    "code": "oas3-api-servers",
    "message": "OpenAPI \\"servers\\" must be present and non-empty array.",
    "range": Array [
      Object {
        "character": 0,
        "line": 0,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "oas3-schema",
    "message": "Object must have required property \\"paths\\".",
    "range": Array [
      Object {
        "character": 0,
        "line": 0,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Hint",
    "source": "spectral",
  },
  Object {
    "code": "info-description",
    "message": "Info \\"description\\" must be present and non-empty string.",
    "range": Array [
      Object {
        "character": 6,
        "line": 2,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "oas3-schema",
    "message": "\\"info\\" property type must be object.",
    "range": Array [
      Object {
        "character": 6,
        "line": 2,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Hint",
    "source": "spectral",
  },
]
`;

exports[`Workspace, basic ruleset negative globs Invalid files trigger generation of diagnostics but only if not ignored simple-ignore.yaml 1`] = `Array []`;

exports[`Workspace, basic ruleset negative globs Invalid files trigger generation of diagnostics but only if not ignored template.cfn.yaml 1`] = `Array []`;
